src/chunk_constraint.c

```diff
@@ -802,7 +814,17 @@ chunk_constraint_add(HeapTuple constraint_tuple, void *arg)
 
 	if (chunk_constraint_need_on_chunk(cc->chunk_relkind, constraint))
 	{
-		ts_chunk_constraints_add(cc->ccs, cc->chunk_id, 0, NULL, NameStr(constraint->conname));
+		bool correlated = false;
+
+		/* check if it's a correlated constraint */
+		if (strncmp(CC_DIM_PREFIX, NameStr(constraint->conname), strlen(CC_DIM_PREFIX)) == 0)
+			correlated = true;
```

Comment:
Another example where a dimension type would make sense.

Reply:
@erimatnor Ok done. Removed the `correlated` boolean from the dimension catalog table and introduced a `type "char"` column to represent the various types of dimensions now.